### PR TITLE
optimize: optimize read of storage configuration

### LINKED ISSUE #4916: seata 通过docker 部署参数兼容问题
seata 1.5.0 版本中使用docker 或 docker-compose 部署时，指定环境变量 -STORE_MODE 参数时，如果参数使用大写会导致seata启动失败，无明显提示，比如：提示未知的mode 或者 同时兼容大小写配置

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

- [x] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did

optimize read of storage configuration

1. `ParameterParser`  only responsible for startup parameters
2. `storeMode,  sessionMode, lockMode`  aggregated into  `StoreConfig`,  to avoid using `getConfig` everywhere.
3.  `storeMode,  sessionMode, lockMode` ignore case.
4.  resolved  not recognized license by SW-eyes

### Ⅱ. Does this pull request fix one issue?

fixes #4916 #4923 #4937

### CODE REVIEW

File: server/src/main/java/io/seata/server/store/StoreConfig.java [hunk 1]
```diff
@@ -15,20 +15,44 @@
...
 import static io.seata.core.constants.ConfigurationKeys.STORE_FILE_PREFIX;
 
-
 /**
  * @author lizhao
  */
 public class StoreConfig {
 
     private static final Configuration CONFIGURATION = ConfigurationFactory.getInstance();
+    private static StoreMode storeMode;
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/store/StoreConfig.java:34]: 都加一下volatile吧,避免那里有异步线程在读取

**Comment by robynron** [comment]
LGTM

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by robynron — APPROVED** [review]
LGTM

**Pull request merged**